.github/workflows/_codeql.yaml

```diff
@@ -0,0 +1,53 @@
+name: "CodeQL"
+
+on:
+  workflow_call:
+    inputs:
+      codeql:
+        description: "Specify the trigger type"
+        required: true
+        type: string
+        default: "push"
+    secrets:
+      codeql_token:
+        description: "Token for CodeQL"
+        required: true
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/using-larger-runners (GitHub.com only)
+    runs-on: "ubuntu-latest"
+    timeout-minutes: 120
+    permissions:
+      # required for all workflows
+      security-events: write
+
+      # required to fetch internal or private CodeQL packs
+      packages: read
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - language: python
+            build-mode: none
+        # To learn more about changing the languages that are analyzed or customizing the build mode for your analysis,
+        # see https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/customizing-your-advanced-setup-for-code-scanning.
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: ${{ matrix.language }}
+          build-mode: ${{ matrix.build-mode }}
```

Comment:
matrix?

Reply:
this is for it we were using multiple languages
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#using-a-matrix-strategy

Reply:
and we have the strategy using matrix on lines 34...

Reply:
Ok, but as there is a single entry in the matrix I suggest you remove it and put the variables here

Reply:
this was the template provided, isn't just simpler to keep it?